src/components/Table/BaseTable.tsx

```diff
@@ -51,127 +55,171 @@ type SpruceTableProps = {
+                            }
+                          />
+                        ) : (
+                          <TableSearchPopover
+                            data-cy={meta?.search?.["data-cy"]}
+                            onConfirm={(value) =>
+                              header.column.setFilterValue(value)
+                            }
+                            placeholder={meta?.search?.placeholder}
+                            value={
+                              (header?.column?.getFilterValue() as string) ?? ""
+                            }
+                          />
+                        ))}
+                    </HeaderCell>
+                  );
+                })}
+              </HeaderRow>
             ))}
-            {row.original.renderExpandedContent && (
-              <StyledExpandedContent row={row} />
+          </TableHead>
+          <TableBody>
+            {loading && (
+              <TableLoader
+                numColumns={table.getAllColumns().length}
+                numRows={loadingRows}
+              />
             )}
-            {row.subRows &&
-              row.subRows.map((subRow) => (
-                <Row
-                  key={subRow.id}
-                  row={subRow}
-                  className={css`
-                    &[aria-hidden="false"] td > div[data-state="entered"] {
-                      max-height: unset;
-                    }
-                  `}
-                >
-                  {subRow.getVisibleCells().map((cell) => (
-                    <Cell key={cell.id}>
-                      {flexRender(
-                        cell.column.columnDef.cell,
-                        cell.getContext()
-                      )}
-                    </Cell>
-                  ))}
-                </Row>
-              ))}
-          </Row>
+            <RenderRows rows={rows} virtualRows={virtualRows} />
+          </TableBody>
+        </StyledTable>
+        {!loading &&
+          rows.length === 0 &&
+          (emptyComponent || "No data to display")}
+      </>
+    );
+  }
+);
+
+const RenderRows = <T extends LGRowData>({
+  rows,
+  virtualRows,
+}: {
+  virtualRows?: VirtualItem[];
+  rows: LeafyGreenTableRow<LGTableDataType<T>>[];
+}) => {
+  const hasVirtualRows = virtualRows && virtualRows.length > 0;
+  return (
+    <>
+      {!hasVirtualRows &&
+        rows.map((row) => (
+          <RenderableRow row={row} key={row.id} virtualRow={null} />
         ))}
-      </TableBody>
-    </StyledTable>
+      {hasVirtualRows &&
+        virtualRows.map((vr) => {
+          const row = rows[vr.index];
+          return <RenderableRow row={row} key={row.id} virtualRow={vr} />;
+        })}
+    </>
+  );
+};
 
-    {!loading &&
-      table.getRowModel().rows.length === 0 &&
-      (emptyComponent || "No data to display")}
-  </>
+const RenderableRow = <T extends LGRowData>({
+  row,
+  virtualRow,
+}: {
+  row: LeafyGreenTableRow<LGTableDataType<T>>;
```

Comment:
I think the typing here should just be
```suggestion
  row: LeafyGreenTableRow<T>;
```
(see [LeafyGreen](https://github.com/mongodb/leafygreen-ui/blob/9a35ec367bb88f471082d58c535937e2b2be9d7f/packages/table/src/ExpandedContent/ExpandedContent.spec.tsx#L39))